### PR TITLE
DOC: Clarify check_like behavior in assert_frame_equal

### LINKED ISSUE #22052: pd.testing.assert_frame_equal check_like not working like expected
#### Code Sample

```python
pd.testing.assert_frame_equal(
pd.DataFrame([{'filename':'a'}, {'filename': 'b'}]),
pd.DataFrame([{'filename':'b'}, {'filename': 'a'}]),
check_like=True)

AssertionError: DataFrame.iloc[:, 0] are different

DataFrame.iloc[:, 0] values are different (100.0 %)
[left]:  [a, b]
[right]: [b, a]
```
#### Problem description
According to the [documentation (version 0.23.3) ](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.testing.assert_frame_equal.html#pandas.testing.assert_frame_equal),  pandas.testing.assert_frame_equal takes a "check_like" parameter which can be set to True if the function should ignore the order of columns & rows. 

>check_like : bool, default False
>If true, ignore the order of rows & columns

This does not work as I expect it to. When creating two Dataframe using a dict, it asserts them as being different due to the order of the rows.

#### Expected Output
I would expect the test to pass.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.3.final.0
python-bits: 64
OS: Linux
OS-release: 4.9.93-linuxkit-aufs
machine: x86_64
processor:
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.23.3
pytest: 3.6.2
pip: 9.0.1
setuptools: 33.1.1
Cython: None
numpy: 1.15.0
scipy: None
pyarrow: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: None
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.999999999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by DeanLa**
- [x] closes #22052

### CODE REVIEW

File: pandas/util/testing.py [hunk 1]
```diff
@@ -1348,7 +1348,9 @@ def assert_frame_equal(left, right, check_dtype=True,
     check_categorical : bool, default True
         Whether to compare internal Categorical exactly.
     check_like : bool, default False
-        If true, ignore the order of rows & columns
+        If true, ignore the order of rows & columns.
```
Comment (WillAyd) [inline pandas/util/testing.py]: Can you capitalize True?

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by DeanLa** [comment]
Changed as requested

File: pandas/util/testing.py [hunk 2]
```diff
@@ -1348,7 +1348,9 @@ def assert_frame_equal(left, right, check_dtype=True,
     check_categorical : bool, default True
         Whether to compare internal Categorical exactly.
     check_like : bool, default False
-        If true, ignore the order of rows & columns
+        If True, ignore the order of index & columns.
+        Note: row labels must match their respective rows
```
Comment (WillAyd) [inline pandas/util/testing.py]: Use `index` instead of `row`

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. @WillAyd merge when satisfied.

**Comment by WillAyd** [comment]
Thanks @DeanLa !

**Pull request merged**